src/Tools/ExternalAccess/Razor/RazorAnalyzerAssemblyResolver.cs

```diff
@@ -37,6 +39,14 @@ internal sealed class RazorAnalyzerAssemblyResolver() : IAnalyzerAssemblyResolve
                         return loadedAssembly;
                     }
 
+                    // Depending on who wins the race to load these assemblies, the base directory will either be the tooling root (if Roslyn wins)
+                    // or the ServiceHubCore subfolder (razor). In the root directory these are netstandard2.0 targeted, in ServiceHubCore they are 
+                    // .net targeted. We need to always pick the same set of assemblies regardless of who causes us to load. Because this code only
+                    // runs in a .net based host, it's safe to always choose the .net targeted ServiceHubCore versions.
+                    if (!rootDirectory.EndsWith(ServiceHubCoreFolderName))
```

Comment:
Are we sure `rootDirectory` doesn't end with `/` or `\`? Shouldn't we trim that before checking `EndsWith`?

Reply:
It doesn't in my testing, but it seems fair to trim it just in case.